### PR TITLE
seed contrib.timeseries with a IndependentMaternGP model

### PR DESCRIPTION
**Pull request by martinjankowiak**
addresses #2090

### CODE REVIEW

File: tests/contrib/timeseries/test_gp.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    assert gp_log_prob.shape == (obs_dim,)
+
+    times = dt * torch.arange(T).double()
+
+    for dim in range(obs_dim):
+        lengthscale = gp.kernel.log_length_scale.exp()[dim]
+        variance = (2.0 * gp.kernel.log_kernel_scale).exp()[dim]
+        obs_noise = (2.0 * gp.log_obs_noise_scale).exp()[dim]
+
+        kernel = pyro.contrib.gp.kernels.Matern32 if nu == 1.5 else pyro.contrib.gp.kernels.Matern52
```
Comment (martinjankowiak) [inline tests/contrib/timeseries/test_gp.py]: this tests the correctness of `MaternKernel` building blocks

File: pyro/contrib/timeseries/gp.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
+import torch
+import torch.nn as nn
+from torch.distributions import MultivariateNormal
+
+import pyro.distributions as dist
+from pyro.ops.ssm_gp import MaternKernel
+
+
+class TimeSeriesModel(nn.Module):
```
Comment (fritzo) [inline pyro/contrib/timeseries/gp.py]: Can you move this to a separate file, say named `base.py`?

**Comment by fritzo** [comment]
We should also add a docs page.

File: pyro/contrib/timeseries/gp.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            of shape `(T, obs_dim)`, where `T` is the length of the time series and `obs_dim`
+            is the dimension of the real-valued `targets` at each time step
+        :returns torch.Tensor: A 0-dimensional log probability for the case of properly
+            multivariate time series models in which the output dimensions are correlated;
+            otherwise returns a 1-dimensional tensor of log probabilities for batched
+            univariate time series models.
+        """
+        raise NotImplementedError
+
+    def predict(self, targets, dts):
```
[the remark this answers is not in the document]
Comment (martinjankowiak) [reply pyro/contrib/timeseries/gp.py]: ok

File: pyro/contrib/timeseries/gp.py (hunk 2 above)
Comment (martinjankowiak) [reply pyro/contrib/timeseries/gp.py]: ok

File: docs/source/contrib.timeseries.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Time Series
```
Comment (fritzo) [inline docs/source/contrib.timeseries.rst]: Also register this file in index.rst:
```diff
diff --git a/docs/source/index.rst b/docs/source/index.rst
index 3e682fd0..d443a01c 100644
--- a/docs/source/index.rst
+++ b/docs/source/index.rst
@@ -36,6 +36,7 @@ Pyro Documentation
    contrib.gp
    contrib.minipyro
    contrib.oed
+   contrib.timeseries
    contrib.tracking
```

Comment (martinjankowiak) [reply docs/source/contrib.timeseries.rst]: done

File: pyro/contrib/timeseries/gp.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            represent the training data that are conditioned on for the purpose of making
+            forecasts.
+        :param torch.Tensor dts: A 1-dimensional tensor of times to forecast into the future,
+            with zero corresponding to the time of the final target `targets[-1]`.
+        :returns torch.distributions.Normal: Returns a predictive Normal distribution with batch shape `(S,)` and
+            event shape `(obs_dim,)`, where `S` is the size of `dts`.
+        """
+        filtering_state = self._filter(targets)
+        predicted_mean, predicted_covar = self._forecast(dts, filtering_state)
+        return torch.distributions.Normal(predicted_mean, predicted_covar.sqrt())
```
Comment (martinjankowiak) [inline pyro/contrib/timeseries/gp.py]: do we want `to_event(1)` here?

File: tests/ops/test_ssm_gp.py [hunk 6]
```diff
@@ -16,9 +16,13 @@ def test_matern_kernel(num_gps, nu):
...
-    assert_equal(forward_backward,
-                 torch.eye(mk.state_dim).unsqueeze(0).expand(num_gps, mk.state_dim, mk.state_dim))
+    eye = torch.eye(mk.state_dim).unsqueeze(0).expand(num_gps, mk.state_dim, mk.state_dim)
+    assert_equal(forward_backward, eye)
 
     # let's just check that these are PSD
     mk.stationary_covariance().cholesky()
     mk.process_covariance(forward).cholesky()
+
+    # evolving forward infinitesimally should yield the identity
```
Comment (martinjankowiak) [inline tests/ops/test_ssm_gp.py]: added a test
